api/scpca_portal/readme_file.py

```diff
@@ -12,6 +14,112 @@
 TEMPLATE_ROOT = settings.TEMPLATE_PATH / "readme"
 TEMPLATE_FILE_PATH = TEMPLATE_ROOT / "readme.md"
 
+# Dataset Readme Templates
+README_ROOT = settings.TEMPLATE_PATH / "dataset_readme"
+HEADER_TEMPLATE = README_ROOT / "header/index.md"
+USAGE_TEMPLATE = README_ROOT / "usage/index.md"
+CHANGELOG_TEMPLATE = README_ROOT / "changelog/index.md"
+CONTACT_TEMPLATE = README_ROOT / "contact/index.md"
+CITATION_TEMPLATE = README_ROOT / "citation/index.md"
+TERMS_TEMPLATE = README_ROOT / "terms_of_use/index.md"
+
+
+# Contents
+DATASET_CONTENTS_ROOT = README_ROOT / "contents"
+
+
+def merge_partials(partials: list[str]):
+    """
+    Takes a list of rendered templates, strips, removes empty and combines with new line.
+    """
+    # For the conte
+    readme_partials: list[str] = list(filter(None, [p.strip() for p in partials]))
+
+    return "\n".join(readme_partials)
+
+
+def get_contents_dict(dataset) -> dict:
+    """
+    Takes dataset instance
+    returns dictionary that relates the content template to the projects that it applies to.
+    ex:
+    {
+        "SINGLE_CELL_EXPERIMENT_SINGLE_CELL.md": ["SCPCP0000000"],
+        "SINGLE_CELL_EXPERIMENT_SINGLE_CELL_MERGED.md": ["SCPCP111111"],
+        "SINGLE_CELL_EXPERIMENT_SPATIAL.md": ["SCPCP222222"],
+    }
+    """
+
+    body = defaultdict(set)
+
+    # metadata can't be combined
+    if dataset.format == DatasetFormats.METADATA:
+        body["METADATA.md"] = dataset.data.keys()
+        return body
+
+    # project data types
+    for project_id, project_options in dataset.data.items():
+        if single_cell_samples := dataset.get_samples(project_id, Modalities.SINGLE_CELL):
+            # template are named "<DatasetFormats>_<Modalities>_<MERGED|MULTIPLEXED?>.md"
```

Comment:
```py
    # template are named "<DatasetFormats>_<Modalities>_<MERGED|MULTIPLEXED?>.md"
```

I was thinking that since we use this naming pattern for readme templates, we could potentially adopt the same pattern for the `CCDLDatasetNames` enum members as well, to improve consistency. Let me know your thoughts on this!

e.g. ) Using the `<DatasetFormats>_<Modalities>_<OTHER_OPTION>`  pattern:
From `CCDLDatasetNames.SINGLE_CELL_EXPERIMENT_SINGLE_CELL` to `CCDLDatasetNames.SINGLE_CELL_EXPERIMENT_SINGLE_CELL`